Virtual Network Gateway (VPN) regression starting in AzureRM 4.10.0

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.9.8

### AzureRM Provider Version

4.10.0

### Affected Resource(s)/Data Source(s)

azurerm_virtual_network_gateway

### Terraform Configuration Files

```hcl
resource azurerm_virtual_network_gateway vpn {
  name                = var.vpnGateway.name
  resource_group_name = data.azurerm_virtual_network.studio.resource_group_name
  location            = data.azurerm_virtual_network.studio.location
  type                = "Vpn"
  sku                 = var.vpnGateway.tier
  vpn_type            = var.vpnGateway.type
  generation          = var.vpnGateway.generation
  enable_bgp          = var.vpnGateway.enableBgp
  active_active       = var.virtualNetwork.gateway.ipAddress2.name != ""
  ip_configuration {
    name                 = "ipConfig1"
    subnet_id            = data.azurerm_subnet.gateway.id
    public_ip_address_id = data.azurerm_public_ip.gateway1.id
  }
  dynamic ip_configuration {
    for_each = var.virtualNetwork.gateway.ipAddress2.name != "" ? [1] : []
    content {
      name                 = "ipConfig2"
      subnet_id            = data.azurerm_subnet.gateway.id
      public_ip_address_id = data.azurerm_public_ip.gateway2[0].id
    }
  }
  dynamic vpn_client_configuration {
    for_each = length(var.vpnGateway.pointToSiteClient.addressSpace) > 0 ? [1] : []
    content {
      address_space = var.vpnGateway.pointToSiteClient.addressSpace
      root_certificate {
        name             = var.vpnGateway.pointToSiteClient.rootCertificate.name
        public_cert_data = var.vpnGateway.pointToSiteClient.rootCertificate.data
      }
    }
  }
}
```


### Debug Output/Panic Output

```shell
azurerm_virtual_network_gateway.vpn: Creating...
╷
│ Error: Plugin did not respond
│
│   with azurerm_virtual_network_gateway.vpn,
│   on gateway.tf line 23, in resource "azurerm_virtual_network_gateway" "vpn":
│   23: resource azurerm_virtual_network_gateway vpn {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-azurerm_v4.11.0_x5.exe plugin:

panic: interface conversion: interface {} is *schema.Set, not []interface {}

goroutine 71 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/network.expandVirtualNetworkGatewayVpnClientConfig(0xde4bb40?, {{0xc000a380c0, 0x24}, {0xc002c78450, 0x25}, {0xc002e409e0, 0xb}})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/network/virtual_network_gateway_resource.go:1096 +0x1ae5
github.com/hashicorp/terraform-provider-azurerm/internal/services/network.getVirtualNetworkGatewayProperties({{0xc000a380c0, 0x24}, {0xc002c78450, 0x25}, {0xc002e409e0, 0xb}}, 0xc002cb7e00)
        github.com/hashicorp/terraform-provider-azurerm/internal/services/network/virtual_network_gateway_resource.go:935 +0x7bb
github.com/hashicorp/terraform-provider-azurerm/internal/services/network.resourceVirtualNetworkGatewayCreate(0xc002cb7e00, {0x6dfe0e0, 0xc001e00908})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/network/virtual_network_gateway_resource.go:675 +0x3fc
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x86c0988?, {0x86c0988?, 0xc002e4b980?}, 0xd?, {0x6dfe0e0?, 0xc001e00908?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:794 +0x15f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc001664620, {0x86c0988, 0xc002e4b980}, 0xc000de45b0, 0xc002cb7c80, {0x6dfe0e0, 0xc001e00908})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:937 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000377db8, {0x86c0988?, 0xc002e4b8c0?}, 0xc002e640a0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/grpc_provider.go:1155 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc0002e4f80, {0x86c0988?, 0xc002e4b5f0?}, 0xc002e640a0)
        github.com/hashicorp/terraform-plugin-mux@v0.17.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001256320, {0x86c0988?, 0xc002e4ac30?}, 0xc000794c40)
        github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov5/tf5server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x7ba7300, 0xc001256320}, {0x86c0988, 0xc002e4ac30}, 0xc002cb7380, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:611 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001cee00, {0x86c0988, 0xc002e4aba0}, {0x86eec80, 0xc00159a000}, 0xc002e49200, 0xc001d07da0, 0xddb4a78, 0x0)
        google.golang.org/grpc@v1.67.1/server.go:1394 +0xe49
google.golang.org/grpc.(*Server).handleStream(0xc0001cee00, {0x86eec80, 0xc00159a000}, 0xc002e49200)
        google.golang.org/grpc@v1.67.1/server.go:1805 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.67.1/server.go:1029 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 52
        google.golang.org/grpc@v1.67.1/server.go:1040 +0x125

Error: The terraform-provider-azurerm_v4.11.0_x5.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

_No response_

### Actual Behaviour

_No response_

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/network/virtual_network_gateway_resource.go
+++ b/internal/services/network/virtual_network_gateway_resource.go
@@ -1093,7 +1093,7 @@ func expandVirtualNetworkGatewayVpnClientConfig(d *pluginsdk.ResourceData, vnetG
 		addresses = append(addresses, addr.(string))
 	}
 
-	rootCertsConf := conf["root_certificate"].([]interface{})
+	rootCertsConf := conf["root_certificate"].(*pluginsdk.Set).List()
 	rootCerts := make([]virtualnetworkgateways.VpnClientRootCertificate, 0, len(rootCertsConf))
 	for _, rootCertSet := range rootCertsConf {
 		rootCert := rootCertSet.(map[string]interface{})
@@ -1106,7 +1106,7 @@ func expandVirtualNetworkGatewayVpnClientConfig(d *pluginsdk.ResourceData, vnetG
 		rootCerts = append(rootCerts, r)
 	}
 
-	revokedCertsConf := conf["revoked_certificate"].([]interface{})
+	revokedCertsConf := conf["revoked_certificate"].(*pluginsdk.Set).List()
 	revokedCerts := make([]virtualnetworkgateways.VpnClientRevokedCertificate, 0, len(revokedCertsConf))
 	for _, revokedCertSet := range revokedCertsConf {
 		revokedCert := revokedCertSet.(map[string]interface{})

--- a/internal/services/network/virtual_network_gateway_resource_test.go
+++ b/internal/services/network/virtual_network_gateway_resource_test.go
@@ -852,7 +852,7 @@ resource "azurerm_virtual_network_gateway" "test" {
 
   vpn_client_configuration {
     address_space        = ["10.2.0.0/24"]
-    vpn_client_protocols = ["IkeV2"]
+    vpn_client_protocols = ["OpenVPN"]
 
     radius_server_address = "1.2.3.4"
     radius_server_secret  = "1234"
@@ -872,6 +872,39 @@ resource "azurerm_virtual_network_gateway" "test" {
       ike_integrity             = "SHA256"
       dh_group                  = "DHGroup14"
       pfs_group                 = "PFS14"
+    }
+
+    vpn_auth_types = ["Radius", "Certificate"]
+
+    revoked_certificate {
+      name       = "third"
+      thumbprint = "83FFBFC8848B5A5836C94D0112367E16148A286F"
+    }
+
+    root_certificate {
+      name             = "DigiCert-CA-Renamed"
+      public_cert_data = <<EOF
+MIIDuzCCAqOgAwIBAgIQCHTZWCM+IlfFIRXIvyKSrjANBgkqhkiG9w0BAQsFADBn
+MQswCQYDVQQGEwJVUzEVMBMGA1UEChMMRGlnaUNlcnQgSW5jMRkwFwYDVQQLExB3
+d3cuZGlnaWNlcnQuY29tMSYwJAYDVQQDEx1EaWdpQ2VydCBGZWRlcmF0ZWQgSUQg
+Um9vdCBDQTAeFw0xMzAxMTUxMjAwMDBaFw0zMzAxMTUxMjAwMDBaMGcxCzAJBgNV
+BAYTAlVTMRUwEwYDVQQKEwxEaWdpQ2VydCBJbmMxGTAXBgNVBAsTEHd3dy5kaWdp
+Y2VydC5jb20xJjAkBgNVBAMTHURpZ2lDZXJ0IEZlZGVyYXRlZCBJRCBSb290IENB
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAvAEB4pcCqnNNOWE6Ur5j
+QPUH+1y1F9KdHTRSza6k5iDlXq1kGS1qAkuKtw9JsiNRrjltmFnzMZRBbX8Tlfl8
+zAhBmb6dDduDGED01kBsTkgywYPxXVTKec0WxYEEF0oMn4wSYNl0lt2eJAKHXjNf
+GTwiibdP8CUR2ghSM2sUTI8Nt1Omfc4SMHhGhYD64uJMbX98THQ/4LMGuYegou+d
+GTiahfHtjn7AboSEknwAMJHCh5RlYZZ6B1O4QbKJ+34Q0eKgnI3X6Vc9u0zf6DH8
+Dk+4zQDYRRTqTnVO3VT8jzqDlCRuNtq6YvryOWN74/dq8LQhUnXHvFyrsdMaE1X2
+DwIDAQABo2MwYTAPBgNVHRMBAf8EBTADAQH/MA4GA1UdDwEB/wQEAwIBhjAdBgNV
+HQ4EFgQUGRdkFnbGt1EWjKwbUne+5OaZvRYwHwYDVR0jBBgwFoAUGRdkFnbGt1EW
+jKwbUne+5OaZvRYwDQYJKoZIhvcNAQELBQADggEBAHcqsHkrjpESqfuVTRiptJfP
+9JbdtWqRTmOf6uJi2c8YVqI6XlKXsD8C1dUUaaHKLUJzvKiazibVuBwMIT84AyqR
+QELn3e0BtgEymEygMU569b01ZPxoFSnNXc7qDZBDef8WfqAV/sxkTi8L9BkmFYfL
+uGLOhRJOFprPdoDIUBB+tmCl3oDcBy3vnUeOEioz8zAkprcb3GHwHAK+vHmmfgcn
+WsfMLH4JCLa/tRYL+Rw/N3ybCkDp00s0WUZ+AoDywSl0Q/ZEnNY0MsFiw6LyIdbq
+M/s/1JRtO3bDSzD9TazRVzn2oBqzSa8VgIo5C1nOnoAKJTlsClJKvIhnRlaLQqk=
+EOF
     }
   }
 }

